apiserver/facades/client/applicationoffers/register.go

```diff
@@ -20,12 +21,12 @@ import (
 // Register is called to expose a package of facades onto a given registry.
 func Register(registry facade.FacadeRegistry) {
 	registry.MustRegister("ApplicationOffers", 5, func(stdCtx context.Context, ctx facade.ModelContext) (facade.Facade, error) {
-		return newOffersAPI(ctx)
+		return newOffersAPI(stdCtx, ctx)
 	}, reflect.TypeOf((*OffersAPIv5)(nil)))
 }
 
 // newOffersAPI returns a new application offers OffersAPI facade.
-func newOffersAPI(facadeContext facade.ModelContext) (*OffersAPIv5, error) {
+func newOffersAPI(ctx context.Context, facadeContext facade.ModelContext) (*OffersAPIv5, error) {
```

Comment:
@hpidcock and I were discussing this problem the other day where we now have situations where we need to pass a std context and new has the possibility of returning an error.

Where we have landed is changing the name from new to make. For example this should become `makeOffersAPI`. Generally new funcs shouldn't be blowing up with errors if we can avoid it. I think lets change it to a make variant which describes the job a bit better.